### DISCUSSION TITLE
Antialiasing (only on Linux?)

### DISCUSSION THREAD

**Discussion by vrafaeli** [friction2d/friction#147] [Q&A]
I use Enve's latest 2022 build regularly and this issue has persisted since.

Apparently native vector graphics still have antialiasing problem, at least in the builds I use (Linux appimage, for both Friction and the Enve).

This is rendered from the 0.9.6 beta 1. Needs to viewed at 100%, not zoomed out.

![aliasing issues rendered from 0_9_6beta 00001](https://github.com/friction2d/friction/assets/70129529/c88d6c6d-99ab-4a27-a634-4605ecd7138e)

I used to cheat my way out of these jaggy lines in Enve by assigning thin outline with MyPaint brushes so my assets may have nice edges. Now that the brushes have been deprecated I don't really mind the direction… but my "cheat" has now been disabled.

To sum it up, native geometry (rectangle, ellipse) has smooth edges only when _not_ assigned outline. ✅

Native path also seems to have smooth edges when _not_ outlined. ✅

Native text object does have issue even when not outlined. When converted to path, the issue persists.

While we are at this, exporting PNG still images from the Render dock also seems to refuse single frame; I had to assign 3 frames for the screenshot above.

This is a dealbreaker because it's so basic. I really don't want to go back to Natron only because of these. Seems like a Linux-driver issue since no one ever complained…? idk

On the bright side, color gradients seem to have improved a bit, or even quite noticeably, since Enve. And the workspace tweak is great; I like having two docks flanking the viewport so I can have a more balanced judgment on what's going on on the canvas proportion-wise. Overall it looks cleaner now, and the appimage memorizes my last layout, unlike Enve, which I appreciate a lot.

Machine is AMD A6-6400K with discreet RX550 card 4GB running PopOS 20.04 if you guys decide to take a look at this.

**Comment by rodlie** [comment]
Hi,

Looks ok to me (render looks the same):
![147](https://github.com/friction2d/friction/assets/34516798/b9c77e78-d73c-4545-b150-689c9a5e8b29)

Ubuntu 22.04 with Intel 620 iGPU, this is the main dev machine for Friction.

> While we are at this, exporting PNG still images from the Render dock also seems to refuse single frame; I had to assign 3 frames for the screenshot above.

Thanks for reporting, seems like the queue won't start if under 3 frames, will take a look at the code.

**Comment by rodlie** [comment]
> Machine is AMD A6-6400K with discreet RX550 card 4GB running PopOS 20.04 if you guys decide to take a look at this.

Just tested PopOS 20.04 live iso with AppImage and no issues here (on Intel).

Probably an issue with the Radeon driver(?). Try something (live iso) with a newer Mesa stack and kernel. If you run Wayland try in X11.

Could of course be an issue with Friction and Radeon in general, I don't test Radeon cards at all (only Intel and NVIDIA).

**Comment by rodlie** [comment]
> While we are at this, exporting PNG still images from the Render dock also seems to refuse single frame; I had to assign 3 frames for the screenshot above.

Try this: https://sourceforge.net/projects/friction/files/0.9.6-d292ce76/Friction-0.9.6-d292ce76-x86_64.AppImage/download

**Reply by vrafaeli** [reply]
This seems to solve the single frame assignment; tried a few runs with different project files and changed codecs too (png, tiff). Single frame stills came out okay. 👍

**Reply by rodlie** [reply]
Awesome, then the issue was in ffmpeg.

**Comment by vrafaeli** [comment]
.
Hi @rodlie, thank you very much for the response, will check out the new appimage.
Re the main issue I remember something for what it's worth.

There was one time when working using Enve I needed to go back to the one installed on Windows 7 just because I couldn't find a way to smooth out the rough edges, the render result from Linux was unacceptable and it was a deliverables for a client. This Enve on Win7 had no jaggy problem and was not even the latest version at the time. The same computer.

That might narrow it down? At least we know the Radeon card is normal when not on Linux.

But I have also worked with vector graphics using Inkscape, Gimp (path object), tested out Synfig a couple times, Figma online & a few of its competitors, took a peek at Lottie the other day, and lately revisiting Natron (roto object) all on the same PopOS. Mostly no issues.
A slightly rough edge can be seen in Natron's native text object. But generally is only found in Enve / Friction and Opentoonz / Tahoma2d. I will try a live iso with other version or distro.

.

**Reply by rodlie** [reply]
> That might narrow it down? At least we know the Radeon card is normal when not on Linux.

All we know is that the Radeon card you have and the driver stack on the distro you use does not work with Friction. I do not have any Radeon cards to test, so can't really do much about it.

> But I have also worked with vector graphics using Inkscape, Gimp (path object), tested out Synfig a couple times, Figma online & a few of its competitors, took a peek at Lottie the other day, and lately revisiting Natron (roto object) all on the same PopOS. Mostly no issues.

Can't really compare. Inkscape, Gimp, Synfig, Natron (raster app, not vector) does not use the GPU.

**Comment by dadoprom** [comment]
Hi, try to change in the Acceleration preferences to CPU only, to "Strong CPU preference" and test it... this worked for me with my Radeon Card. I did not have this issue with intel GPU on my laptop.

**Reply by vrafaeli** [reply]
I did try this just yesterday after reading up a bit on what I could find, and I tested it again just now.

What works on my end is I can set the Acceleration all the way to the right (GPU) but check off 'Path GPU acceleration', and the path object is applied a Raster Effects → Blur for around 1 - 1.5 value. This is of course just a band aid. While with the 'Path GPU' checked the blur is still not much help (the softening is there but you can still see the rough, interesting). Guess that'll have to do for now.

What does the Filtering option do? from the top menu bar View → Filtering → none, low, medium etc. 
In Enve it used to help a bit also in this issue; I go for medium while viewport is at 100% res. Here in Friction I haven't noticed any difference.

Would it be helpful if I forward you guys the readings from gpuviewer? My mesa is detected 24.0.6 with X11 I think it's quite recent. Haven't been able to test from any live systems. For whatever reason the Ventoy now gave me kernel panic when booting from different ISOs.

**Reply by vrafaeli** [reply]
Wait… so just to clarify,
I tested various combinations, from strong CPU to strong GPU to the middle/agnostic, it will only work if I disable the Path GPU accel.

I still need to apply the blur effect mentioned above. but with the Path GPU accel on, the jaggy edge is more persistent, if that makes any sense.
·

**Reply by rodlie** [reply]
Path acceleration is the issue (as you probably noticed). Why? I don't know. The problem is in driver and/or Skia (our "engine").

I went through some old computer "spare parts" boxes and found an old RX580, don't know if it works or not yet.

**Reply by rodlie** [reply]
btw, Chrome (and related) uses the same backend as Friction, is vector smooth there?

**Reply by vrafaeli** [reply]
Hi, I tested opening svg files directly on Brave appimage and one other browser.
Everything is crisp and smooth, no issues with antialiasing.

edit: Brave is essentially Chrome, the other browser is Floorp which is a Firefox fork.
·

**Comment by rodlie** [comment]
This has been fixed (with an MSAA option) and will be part of our next release.
